core/src/main/cpp/util/Base64Secure.cpp

```diff
@@ -121,8 +121,17 @@ std::shared_ptr<ByteArray> Base64Secure::decode(const std::string& s) {
     StringConstIterator p = s.begin();
     while (p != s.end()) {
     	const char c = *p++;
+
+    	// Ensure the character is not "negative".
+    	if (c & 0x80) {
+    		invalid = true;
+    		continue;
+    	}
+
+    	// Lookup the character in the decoder map.
```

Comment:
Do you really want to just skip, like we do for whitespace? Shouldn't this be an encoding error? What specifically is wrong with the "negative" char?